### ISSUE TITLE
Lawnmower position on map mirrored along the diagonal between inserted points

### ISSUE THREAD

**Issue by TheNico14** [Thomas55555/husqvarna_automower#441]
**Prerequirements**
[x] I have connected my mower(s) to the Automower Connect App
[x] I have connected my Husqvanra developers account to the Authentication API
[x] I have connected my Husqvarna developers account to the Automower Connect API

**Describe the bug**
I recently tried to add the map and encountered a problem. The position of the lawnmower and the lines drawn seem to be mirrored along the diagonal between the inserted points (top left and bottom right). The position provided by the lawnmower is correct, as it appears exactly on the base when entered on google maps. The position is approximately at coordinates 45.0, 10.0.

**Expected behavior**
I expect to see the robot on the map in the correct position.

**Screenshots**
![map](https://user-images.githubusercontent.com/37806502/236757216-41b6e444-aaab-47c1-ab7e-45e28c58efa6.png)
Red square: lawnmower displayed on the map.
Blue square: actual lawnmower position.

**Comment by prairiesnpr** [comment]
This is a pretty simple function, we just take the mower position in lat long and translate it to a pixel position on the image.  This is done by taking the size of the image and the position of each corner in lat long to come up with a transform. See https://github.com/Thomas55555/husqvarna_automower/blob/92922f5cff3e0168b2a8695af924b3d7ebe17c38/custom_components/husqvarna_automower/camera.py#L191 

Basically, if the mower isn't plotted in the correct position, the most likely issue is that the lat long for the image corners isn't set correctly.   I would double check those coordinates and verify that you haven't reversed the corners.

**Comment by TheNico14** [comment]
Thank you for your reply.

If you can confirm that in the configuration it is correct to enter the top left corner as the first parameter and the bottom right corner as the second, I am sure.
I have checked many times before opening the ticket with google maps the points entered and the position of the robot device tracker and they are in the correct position.

For obvious reasons I can't post the map or my coordinates here, is there any way I can get this information to you privately so I can show you what's going on?

**Comment by prairiesnpr** [comment]
Best to reach out on discord, can DM me at PrairieSnpr#9786.

I wanted to rule out there wasn't an issue with non-negative longitude, so pulled that function out and ran it through a quick sanity check, seems to be working fine in your area.

Grabbed a random yard near the coordinates you posted.

![image](https://github.com/Thomas55555/husqvarna_automower/assets/18637008/87442083-a13c-4a1d-aee1-c0ec3aa03e13)

From google, we get the following coordinates:
``` 
(44.984709298103006, 10.039228629942667) # Bottom Right
(44.98521772246266, 10.038645249542478)   # Top Left
(44.984709298103006, 10.038645249542478) # Bottom Left
(44.98521772246266, 10.039228629942667)   # Top Right
```

Running those through the same function gives the following results.

```
>>> t._scale_to_img(top_left_coord,(200,200))    
(0, 0)
>>> t._scale_to_img(top_right_coord,(200,200)) 
(200, 0)
>>> t._scale_to_img(bottom_left_coord,(200,200)) 
(0, 200)
>>> t._scale_to_img(bottom_right_coord,(200,200)) 
(200, 200)
```

Now if, instead of setting the top_left and bottom_right coordinates correctly, we swap them like so, then rerun our test
```
top_left_coord = (44.984709298103006, 10.039228629942667) #Bottom Right / Reversed
bottom_right_coord = (44.98521772246266, 10.038645249542478) # Top Left / Reversed
```

We see the same result you describe. 

```
>>> t._scale_to_img(top_right_coord,(200,200))      
(0, 200)
>>> t._scale_to_img(bottom_left_coord,(200,200)) 
(200, 0)
```

I think I would swap what you have for the top left and bottom right coordinates and see if that solves your issue.

**Comment by TheNico14** [comment]
After a brief conversation with @prairiesnpr he came to the conclusion that the problem is probably caused by the rotation of my map.
By rotating the map with north upwards the position of the mower is now shown correctly.
Thanks!
